FilmFestivalLoader/IFFR/parse_iffr_html.py

```diff
@@ -508,31 +473,57 @@ def add_screened_film(self):
         finally:
             self.init_screened_film_data()
 
-    def update_film_info(self):
-        # Get the film info of the current film. Its unique existence is guaranteed in AzPageParser.
-        film_info = self.film.film_info(self.iffr_data)
+    def add_paragraph(self):
+        self.article_paragraphs.append(self.article_paragraph)
+        self.article_paragraph = ''
 
+    def set_article(self):
+        self.article = '\n\n'.join(self.article_paragraphs)
+        self.film_info.article = self.article
+
+    def store_combination_from_title(self, combination_title):
+        films = [f for f in self.iffr_data.films if f.title == combination_title]
+        self.store_combination(films[0].filmid)
+
+    def store_combination_from_url(self, url):
+        combination_url = self.repair_url(url)
+        combination_film = self.iffr_data.get_film_by_key(None, combination_url)
+        self.store_combination(combination_film.filmid)
+
+    def store_combination(self, combination_film_id):
+        screened_film_id = self.film.filmid
+        if combination_film_id in Globals.extras_by_main.keys():
+            Globals.extras_by_main[combination_film_id].append(screened_film_id)
+        else:
+            Globals.extras_by_main[combination_film_id] = [screened_film_id]
+
+    @staticmethod
+    def repair_url(url):
+        parts = url.split('/')
+        if parts[4] != url_festival:
+            parts.insert(4, az_url_path.split('/')[2])
+            return '/'.join(parts)
+        return url
+
+    def set_screened_films(self):
         # Set the screened films list of the film info of the current film.
-        film_info.screened_films = self.screened_films
+        self.film_info.screened_films = self.screened_films
         self.print_debug(
             f'SCREENED FILMS of {self.film.title} UPDATED', f'{len(self.screened_films)} screened films added.')
-        self.print_debug(f'SCREENED FILMS LIST of {self.film} is now in info:', f'\n{film_info}')
+        self.print_debug(f'SCREENED FILMS LIST of {self.film} is now in info:', f'\n{self.film_info}')
 
         # Append the film being analysed to the combination programs of the screened films.
         for screened_film in self.screened_films:
             screened_film_infos = [i for i in self.iffr_data.filminfos if i.filmid == screened_film.filmid]
             screened_film_info = screened_film_infos[0]
-            self.print_debug(f'COMBINATION URLS of {screened_film.title}:', f'{screened_film_info.combination_urls}')
-            if len(screened_film_info.combination_urls) == 0:
-                screened_film_info.combination_urls = [self.film.url]
+            combination_urls = screened_film_info.combination_urls
+            if len(combination_urls) == 0:
+                combination_urls = [self.film.url]
             else:
```

Comment:
Please explain why you store URL's, not film ID's.

Reply:
Maybe there used to be a good reason, but I don't see one now.
Changed into a list of films, `compilation_films`, just like `screened_films`.